### ISSUE TITLE
WSL2 Dev Container and file watchers

### ISSUE THREAD

**Issue by njbmartin** [microsoft/vscode-remote-release#3238]
- VSCode Version: 1.46.1
- Local OS Version: Windows 10 2004
- Remote OS Version: Docker with WSL2 Ubunutu 20.04
- Remote Extension/Connection Type: Docker

Steps to Reproduce:

1. Clone https://github.com/njbmartin/gatsby-starter-dev-container
2. Open with Dev Containers on Windows with Docker Desktop WSL2 base engine
3. Install dependencies with `npm i`
4. Start the webserver with `npm start`
5. Make a change to a file within the `src` folder. The project should automatically rebuild (but doesn't).

Does this issue occur when you try this locally?: No
Does this issue occur when you try this locally and all extensions are disabled?: No

This isn't a problem when Docker Desktop doesn't use WSL2. I'm aware of this: https://github.com/microsoft/WSL/issues/4739

**Comment by chrmarti** [comment]
Do you clone the repository on the Windows command line or inside WSL? On a Windows filesystem or a WSL one (what is the path?)?

**Comment by njbmartin** [comment]
Sorry yes it's cloned via Github Desktop, so Windows.

**Comment by chrmarti** [comment]
Is the folder path in `c:\` or `\\wsl$\`? I'm asking because if it is in `c:\` the issue you refer to might cause this.

With WSL2 you will get better I/O performance when you clone to a WSL folder. That is under `\\wsl$\...` on the Windows side. I recommend cloning using the `git` CLI inside WSL, so permissions and line endings of the files are set up for Linux.

**Comment by njbmartin** [comment]
it's under `c:\`. I'll give WSL a proper go.

**Comment by njbmartin** [comment]
Unfortunately, if opening the folder "locally" in VS Code, I get this:

![image](https://user-images.githubusercontent.com/5699471/85905960-d91a9780-b804-11ea-9515-8227259cbee2.png)

**Comment by njbmartin** [comment]
I know I can get it all working if I relied on the CLI, but if there is a GUI, I'd prefer to use that, so yes I'm probably making this harder for myself by not following your advice directly, but I'm trying to get this working the same as an average person who might not use the CLI either.

Found the issue of the above, I had mapped it to a network drive (following recommendations from another site), so VS Code was opening the folder from `U:\`, which translated to `//wsl$/...` instead of `\\wsl$\...` when trying to open in Remote Container or Remote WSL.

Unfortunately from a GUI perspective, this path is initially hidden from Explorer, hence why I mapped it to a network drive for easy access in Windows.

![image](https://user-images.githubusercontent.com/5699471/85907873-ec306600-b80a-11ea-9356-f670dbafca30.png)
